drizzlepac/haputils/poller_utils.py

```diff
@@ -1328,14 +1328,9 @@ def build_poller_table(
         if "aperture" in input_table.colnames:
             cols["aperture"] = input_table["aperture"].tolist()
         else:
-            add_col = Column(
-                ["empty_aperture"] * len(usable_datasets), name="aperture", dtype="str"
-            )
-            input_table.add_column(add_col, index=7)
-            poller_dtype += ("str",)
+            cols["aperture"] = ["empty_aperture"] * len(usable_datasets)
```

Comment:
It is possible for cols["aperture"] Line 1331 to have more rows than are remaining in usable_datasets which will ultimately cause a failure.

Reply:
It shoudn't? Are tests still coming back with uneven indices?

Reply:
I was able to find the edge case that you were alluding to: Full poller table with datasets that are not usable. I altered the analyze.py module to return indices (as well as usable_datasets) to retrieve the corresponding aperture values from the input_table/poller_file.